include/RAJA/policy/hip/kernel/Tile.hpp

```diff
@@ -167,44 +166,33 @@ struct HipStatementExecutor<
 
     // compute trip count
     diff_t len = segment.end() - segment.begin();
-    diff_t i = get_hip_dim<BlockDim>(dim3(blockIdx.x,blockIdx.y,blockIdx.z)) * chunk_size;
-
-    // check have chunk
-    if (i < len) {
+    diff_t i = Indexer::template index<diff_t>() * static_cast<diff_t>(chunk_size);
 
-      // Keep copy of original segment, so we can restore it
-      segment_t orig_segment = segment;
+    // Keep copy of original segment, so we can restore it
+    segment_t orig_segment = segment;
 
-      // Assign our new tiled segment
-      segment = orig_segment.slice(i, chunk_size);
+    // Assign our new tiled segment
+    segment = orig_segment.slice(i, static_cast<diff_t>(chunk_size));
 
-      // execute enclosed statements
-      enclosed_stmts_t::exec(data, thread_active);
+    // execute enclosed statements
+    enclosed_stmts_t::exec(data, thread_active && (i < len));
```

Comment:
Should (i < len) be have_work here?

Reply:
Yes, that's what it is here even if its not given a name.